igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -41,48 +40,51 @@ import java.util.concurrent.ExecutorService
 
 @Slf4j
 @RestController
-@ConditionalOnProperty('jenkins.enabled')
 class BuildController {
-    @Autowired
-    JenkinsMasters masters
 
     @Autowired
     ExecutorService executor
 
     @Autowired
+    BuildMasters buildMasters
+
+    @Autowired
     ObjectMapper objectMapper
 
     @RequestMapping(value = '/builds/status/{buildNumber}/{master}/**')
     Map getJobStatus(@PathVariable String master, @PathVariable Integer buildNumber, HttpServletRequest request) {
         def job = (String) request.getAttribute(
             HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split('/').drop(5).join('/')
-        if (!masters.map.containsKey(master)) {
-            throw new MasterNotFoundException("Master '${master}' not found")
-        }
-        Map result = objectMapper.convertValue(masters.map[master].getBuild(job, buildNumber), Map)
-        try {
-            Map scm = objectMapper.convertValue(masters.map[master].getGitDetails(job, buildNumber), Map)
-            if (scm?.action?.lastBuiltRevision?.branch?.name) {
-                result.scm = scm?.action.lastBuiltRevision
-                result.scm = result.scm.branch.collect {
-                    it.branch = it.name.split('/').last()
-                    it
+        if (buildMasters.filteredMap(BuildServiceProvider.JENKINS).containsKey(master)) {
+            Map result = objectMapper.convertValue(buildMasters.map[master].getBuild(job, buildNumber), Map)
+            try {
+                Map scm = objectMapper.convertValue(buildMasters.map[master].getGitDetails(job, buildNumber), Map)
+                if (scm?.action?.lastBuiltRevision?.branch?.name) {
+                    result.scm = scm?.action.lastBuiltRevision
+                    result.scm = result.scm.branch.collect {
+                        it.branch = it.name.split('/').last()
+                        it
```

Comment:
I think for the most part we try to make `return` statements explicit, with the exception of 1 line closures.